src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -0,0 +1,1048 @@
+            {
+                JsonPropertyInfo propertyInfo = typeInfo.CreateJsonPropertyInfo(typeof(string), "test");
+                Assert.Same(expectedOptions, propertyInfo.Options);
+            }
+        }
+
+        [Theory]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(MyClass))]
+        public static void JsonPropertyInfoPropertyTypeIsSetWhenUsingCreateJsonPropertyInfo(Type propertyType)
+        {
+            JsonSerializerOptions options = new();
+            JsonTypeInfo typeInfo = JsonTypeInfo.CreateJsonTypeInfo(typeof(MyClass), options);
+            JsonPropertyInfo propertyInfo = typeInfo.CreateJsonPropertyInfo(propertyType, "test");
+
+            Assert.Equal(propertyType, propertyInfo.PropertyType);
+        }
+
+        [Fact]
+        public static void JsonPropertyInfoPropertyTypeIsSet()
+        {
+            JsonSerializerOptions options = new();
+            JsonTypeInfo typeInfo = options.TypeInfoResolver.GetTypeInfo(typeof(MyClass), options);
+            JsonPropertyInfo propertyInfo = typeInfo.Properties[0];
+            Assert.Equal(typeof(string), propertyInfo.PropertyType);
+        }
+
+        [Theory]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(MyClass))]
+        public static void JsonPropertyInfoNameIsSetAndIsMutableWhenUsingCreateJsonPropertyInfo(Type propertyType)
+        {
+            JsonSerializerOptions options = new();
+            JsonTypeInfo typeInfo = JsonTypeInfo.CreateJsonTypeInfo(typeof(MyClass), options);
+            JsonPropertyInfo propertyInfo = typeInfo.CreateJsonPropertyInfo(propertyType, "test");
+
+            Assert.Equal("test", propertyInfo.Name);
+
+            propertyInfo.Name = "foo";
+            Assert.Equal("foo", propertyInfo.Name);
+
+            Assert.Throws<ArgumentNullException>(() => propertyInfo.Name = null);
+        }
+
+        [Fact]
+        public static void JsonPropertyInfoNameIsSetAndIsMutableForDefaultResolver()
+        {
+            JsonSerializerOptions options = new();
+            JsonTypeInfo typeInfo = options.TypeInfoResolver.GetTypeInfo(typeof(MyClass), options);
+            JsonPropertyInfo propertyInfo = typeInfo.Properties[0];
+
+            Assert.Equal(nameof(MyClass.Value), propertyInfo.Name);
+
+            propertyInfo.Name = "foo";
+            Assert.Equal("foo", propertyInfo.Name);
+
+            Assert.Throws<ArgumentNullException>(() => propertyInfo.Name = null);
+        }
+
+        [Fact]
+        public static void JsonPropertyInfoForDefaultResolverHasNamingPoliciesRulesApplied()
+        {
+            JsonSerializerOptions options = new();
+            options.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
+            JsonTypeInfo typeInfo = options.TypeInfoResolver.GetTypeInfo(typeof(MyClass), options);
+            JsonPropertyInfo propertyInfo = typeInfo.Properties[0];
```

Comment:
```suggestion
            Assert.Equal(1, typeInfo.Properties.Length);
            JsonPropertyInfo propertyInfo = typeInfo.Properties[0];
```
Or whatever length is appropriate.